pkg/controller/computenodeopenstack/computenodeopenstack_controller.go

```diff
@@ -1229,22 +1286,44 @@ func deleteAllBlockerPodFinalizer(r *ReconcileComputeNodeOpenStack, instance *co
 		log.Info(fmt.Sprintf("Post draining job completed for node: %s", node.Name))
 	}
 	if !allNodesPostDrained {
-		return fmt.Errorf("Not all Post NodeDrain jobs finished, reconciling")
+		// if not all Post draining jobs finished, return a StatusReasonAlreadyExists.
+		// AlreadyExists error will be ignored as we want to wait for the job event to
+		// not spam the operator log with too many not needed reconcile requrest.
+		return &errors.StatusError{ErrStatus: metav1.Status{
+			Status: metav1.StatusFailure,
+			Code:   http.StatusConflict,
+			Reason: metav1.StatusReasonAlreadyExists,
+			Details: &metav1.StatusDetails{
+				Group: instance.GroupVersionKind().Group,
+				Kind:  instance.Kind,
+				Name:  instance.Name,
+			},
+			Message: fmt.Sprintf("Not all POST NodeDrain jobs finished"),
+		}}
 	}
 	log.Info(fmt.Sprintf("All Post NodeDrain jobs finished"))
 
-	// 3. Cleanup finished POST drain jobs
+	// 3. Cleanup
 	for _, node := range nodeList.Items {
-		log.Info(fmt.Sprintf("Deleting draining jobs for node: %s", node.Name))
-		err = deleteJobWithName(r.client, r.kclient, instance, node.Name+"-drain-job-pre")
-		if err != nil && !errors.IsNotFound(err) {
+		// If blocker pod finalizer was already deleted, there is nothing extra to do for this node
```

Comment:
I think this may not be needed as it is already checked out in the code calling this function. But having the extra check does not hurt anyway